src/Microsoft.CSharp/tests/RuntimeBinderExceptionTests.cs

```diff
@@ -0,0 +1,70 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.IO;
+using System.Runtime.Serialization.Formatters.Binary;
+using Xunit;
+
+namespace Microsoft.CSharp.RuntimeBinder.Tests
+{
+    public class RuntimeBinderExceptionTests
+    {
+        [Fact]
+        public void NullaryCtor()
+        {
+            RuntimeBinderException rbe = new RuntimeBinderException();
+            Assert.Null(rbe.InnerException);
+            Assert.Empty(rbe.Data);
+            Assert.True((rbe.HResult & 0xFFFF0000) == 0x80130000); // Error from .NET
+            Assert.Contains(rbe.GetType().FullName, rbe.Message); // Localized, but should contain type name.
+        }
+
+        [Fact]
+        public void StringCtor()
+        {
+            string message = "This is a test message.";
+            RuntimeBinderException rbe = new RuntimeBinderException(message);
+            Assert.Null(rbe.InnerException);
+            Assert.Empty(rbe.Data);
+            Assert.True((rbe.HResult & 0xFFFF0000) == 0x80130000); // Error from .NET
+            Assert.Same(message, rbe.Message);
+            rbe = new RuntimeBinderException(null);
+            Assert.Equal(new RuntimeBinderException().Message, rbe.Message);
+        }
+
+
+        [Fact]
+        public void InnerExceptionCtor()
+        {
+            string message = "This is a test message.";
+            Exception inner = new Exception("This is a test exception");
+            RuntimeBinderException rbe = new RuntimeBinderException(message, inner);
+            Assert.Same(inner, rbe.InnerException);
+        }
+
+        [Fact]
+        public void Serialize()
+        {
+            string message = "This is a test message.";
+            Exception inner = new Exception("This is a test exception");
+            RuntimeBinderException source = new RuntimeBinderException(message, inner);
+            RuntimeBinderException rbe;
+            using (var ms = new MemoryStream())
+            {
+                var bf = new BinaryFormatter();
+                bf.Serialize(ms, source);
+                ms.Flush();
+                ms.Seek(0, SeekOrigin.Begin);
+                rbe = (RuntimeBinderException)bf.Deserialize(ms);
+            }
+
+            Assert.NotSame(source, rbe);
+            Assert.Equal(source.Message, rbe.Message);
+            Assert.IsType<Exception>(rbe.InnerException);
+            Assert.Equal(inner.Message, rbe.InnerException.Message);
+            Assert.Equal(source.HResult, rbe.HResult);
+        }
```

Comment:
There's already a helper for this:
https://github.com/dotnet/corefx/blob/master/src/Common/tests/System/Runtime/Serialization/Formatters/BinaryFormatterHelpers.cs#L24-L25